src/Scripting/Core/ObjectFormatter.cs

```diff
@@ -449,83 +471,179 @@ private string FormatPrimitive(object obj, bool quoteStrings, bool includeCodePo
                 return NullLiteral;
             }
 
-            if (obj is bool)
+            if (obj.GetType() == typeof(int))
```

Comment:
@KevinH-MS That would defeat the pattern recognized by JIT. JIT optimizes these away to a single pointer check (and should also optimize additional checks in the following cast). Not that speed would be crucial here, but that's why I opted to use the same pattern everywhere. 

@cston TypeCode is not portable.